src/components/Area.js

```diff
@@ -0,0 +1,693 @@
+          if (typeof this.children[i] === 'object') {
+            this.children[i].render(this, area, p);
+          }
+        }
+      },
+      (area, evt) => {
+        const down = evt.getDown();
+        const up = 0; //evt.getUp();
+        if (up) {
+          this.props.onMouseUp({
+            x: evt.getX(),
+            y: evt.getY(),
+            width: evt.getAreaWidth(),
+            height: evt.getAreaHeight(),
+            button: up,
+          });
+        } else if (down) {
+          this.props.onMouseDown({
+            x: evt.getX(),
+            y: evt.getY(),
+            width: evt.getAreaWidth(),
+            height: evt.getAreaHeight(),
+            button: down,
+          });
+        } else {
+          const buttons = [];
+          const held = evt.getHeld1To64();
+          if (held > 0) {
+            for (let i = 0; i <= 6; i++) {
+              if (held & Math.pow(2, i)) buttons.push(i + 1);
+              if (!(held >> (i + 1))) break;
+            }
+          }
+          this.props.onMouseMove({
+            x: evt.getX(),
+            y: evt.getY(),
+            width: evt.getAreaWidth(),
+            height: evt.getAreaHeight(),
+            buttons,
+          });
+        }
+      },
+      (area, inOut) => {
+        if (inOut === 0) {
+          this.props.onMouseEnter();
+        } else {
+          this.props.onMouseLeave();
+        }
+      },
+      function dragBroken() {},
+      (area, event) => {
+        if (event.getUp()) {
+          return this.props.onKeyUp({
+            key: event.getKey(),
+            extKey: event.getExtKey(),
+            modifierKey: event.getModifier(),
+            modifiers: event.getModifiers(),
+          });
+        } else {
+          return this.props.onKeyDown({
+            key: event.getKey(),
+            extKey: event.getExtKey(),
+            modifierKey: event.getModifier(),
+            modifiers: event.getModifiers(),
+          });
+        }
+      }
+    );
+  }
+
+  getArea() {
+    return this.element;
+  }
+
+  // to prevent TypeError: Cannot read property 'undefined' of undefined
+  // because onMouseMove, ... shouldn't be handled by DesktopComponent
+  update(oldProps, newProps) {}
+
+  render(parent) {
+    this.addParent(parent);
+  }
+}
+
+Area.PropTypes = {
+  ...universalPropTypes,
+  onMouseMove: PropTypes.func,
+  onMouseUp: PropTypes.func,
+  onMouseDown: PropTypes.func,
+  onMouseEnter: PropTypes.func,
+  onMouseLeave: PropTypes.func,
+  onKeyUp: PropTypes.func,
+  onKeyDown: PropTypes.func,
+  children: PropTypes.oneOfType([
+    PropTypes.element,
+    PropTypes.arrayOf(PropTypes.element),
+  ]),
+};
+
+Area.defaultProps = {
+  ...universalDefaultProps,
+  onMouseMove: e => {},
+  onMouseUp: e => {},
+  onMouseDown: e => {},
+  onMouseEnter: area => {},
+  onMouseLeave: area => {},
+  onKeyUp: (area, event) => {},
+  onKeyDown: (area, event) => {},
+};
+
+function fallback(...vals) {
```

Comment:
What exactly is this used for?

Reply:
In `AreaComponent.render` for the transform handing.
Examples:
```js
fallback(rotate[2], '50%', v => v) // = rotate[2], if not set: "50%"
fallback(translate[2], translate[1], v => // = translate[2], if not set: translate[1]
  this.parseSelf(v, p, true)
)
```